zerver/lib/request.py

```diff
@@ -456,7 +456,17 @@ def _wrapped_req_func(
 
             kwargs[func_var_name] = val
 
-        return req_func(request, *args, **kwargs)
+        return_value = req_func(request, *args, **kwargs)
+
+        if isinstance(return_value, JsonDataResponse) and not request_notes.is_webhook_view:
+            ignored_parameters = set(
+                list(request.POST.keys()) + list(request.GET.keys())
+            ).difference(request_notes.processed_parameters)
+
+            if len(ignored_parameters) > 0:
+                return_value.data["ignored_parameters_unsupported"] = list(ignored_parameters)
```

Comment:
How does this work if `has_request_variables` is called multiple times on a request object (we have some helpers that use it)? I think it is correct because this runs at the end and uses a shared `processed_parameters` value, so we'll just d this work twice with each producing the same answer ... but probably deserves a block comment explaining that.

Reply:
This has been updated (with a comment) so that:
- if there are ignored parameters, the `ignored_parameters_unsupported` key is added/updated with the current list value are added to the data
- if there are no ignored parameters, then any `ignored_parameters_unsupported` key is removed.